frontend/controller/actions/group.js

```diff
@@ -914,41 +912,92 @@ export default (sbp('sbp/selectors/register', {
   ...encryptedAction('gi.actions/group/groupProfileUpdate', L('Failed to update group profile.')),
-  ...encryptedAction('gi.actions/group/proposal', L('Failed to create proposal.')),
+  ...encryptedAction('gi.actions/group/proposal', L('Failed to create proposal.'), async (sendMessage, params) => {
+    const { contractID } = params
+    await sendMessage({
+      ...params,
+      hooks: {
+        onprocessed: (message) => {
+          (async () => {
+            const proposalId = message.hash()
+            const state = await sbp('chelonia/contract/state', contractID)
+            const proposal = state.proposals[proposalId]
+            const proposalToSend = { ...extractProposalData(proposal, { proposalId }), status: STATUS_OPEN }
+
+            await sbp('gi.actions/group/notifyProposalStateInGeneralChatRoom', { groupID: contractID, proposal: proposalToSend })
+          })()
+        }
+      }
+    })
+  }),
   ...encryptedAction('gi.actions/group/proposalVote', L('Failed to vote on proposal.'), async (sendMessage, params) => {
-    const state = await sbp('chelonia/contract/state', params.contractID)
-    const data = params.data
+    const { contractID, data } = params
+    const state = await sbp('chelonia/contract/state', contractID)
     const proposalHash = data.proposalHash
     const proposal = state.proposals[proposalHash]
     const type = proposal.data.proposalType
-    let passPayload
-
-    if (data.vote === VOTE_FOR) {
-      passPayload = {}
-      if (oneVoteToPass(state, proposalHash)) {
-        if (type === PROPOSAL_INVITE_MEMBER) {
-          passPayload = await createInvite({
-            contractID: params.contractID,
-            invitee: proposal.data.proposalData.memberName,
-            creatorID: proposal.creatorID,
-            expires: state.settings.inviteExpiryProposal
-          })
-        }
+
+    const willBePassed = oneVoteToPass(state, proposalHash)
+    const willBeFailed = oneVoteToFail(state, proposalHash)
+    const isVoteFor = data.vote === VOTE_FOR
+    const isVoteAgainst = !isVoteFor
+    let passPayload = isVoteFor ? {} : undefined
+    let proposalToSend
+
+    if (willBePassed && isVoteFor) {
+      if (type === PROPOSAL_INVITE_MEMBER) {
+        passPayload = await createInvite({
+          contractID,
+          invitee: proposal.data.proposalData.memberName,
+          creatorID: proposal.creatorID,
+          expires: state.settings.inviteExpiryProposal
+        })
       }
+
+      proposalToSend = { ...extractProposalData(proposal, { proposalId: proposalHash }), status: STATUS_PASSED }
+    } else if (willBeFailed && isVoteAgainst) {
+      proposalToSend = { ...extractProposalData(proposal, { proposalId: proposalHash }), status: STATUS_FAILED }
     }
 
     return sendMessage({
       ...params,
-      data: {
-        ...data,
-        passPayload
+      data: { ...data, passPayload },
+      hooks: {
+        ...params?.hooks,
+        postpublish: (...args) => {
```

Comment:
Please be consistent. Either `'gi.actions/group/proposal'` should use `postpublish`, or this should use `onprocessed`.

Reply:
The reason why I used `onprocessed` hook for `gi.actions/group/proposal` is that at the time of calling this function, the proposal doesn't exist in state. Because we are creating proposal in `gi.actions/group/proposal/process`.

And the reason why I used `postpublish` here, is that I wanted to call `gi.actions/group/notifyProposalStateInGeneralChatRoom` function just after the event is published, and before processed. The reason is that let's imagine the proposal is to remove a user from the group.
We should create two messages in the general chatroom.
- One is to notify that the proposal is closed by approval
- The other is to say the user has left the group
Which one do you think should be created before? Yes, the first message (which tells that the proposal is closed by approval). That's why I used `postpublish` here instead of `onprocessed`.

And I already made a question [here](https://okturtles.slack.com/archives/C0EH7P20Y/p1721189410310829), because sometimes (especially only in Cypress) the second one is created before the first one.

Reply:
What you are saying doesn't make any sense to me. In both cases, `notifyProposalStateInGeneralChatRoom` will get sent and processed by clients after `proposalVote`.

Reply:
`notifyProposalStateInGeneralChatRoom` is sent after `proposalVote` is published. But not sure before or after the `proposalVote` is processed.

Reply:
So what?

Reply:
Please see this [Cypress failure](https://cloud.cypress.io/projects/q6whky/runs/2689/test-results/c9fc0180-0ff5-4107-8ec6-2f8879a4cf65?actions=%5B%5D&browsers=%5B%5D&groups=%5B%5D&isFlaky=%5B%5D&modificationDateRange=%7B%22startDate%22%3A%221970-01-01%22%2C%22endDate%22%3A%222038-01-19%22%7D&orderBy=EXECUTION_ORDER&oses=%5B%5D&specs=%5B%5D&statuses=%5B%7B%22value%22%3A%22FAILED%22%2C%22label%22%3A%22FAILED%22%7D%5D&testingTypesEnum=%5B%5D&utm_source=github).
I thought I fixed this error in Cypress test case, but it still happens. But this shows what I mean. The order of two messages is random so the Cypress fails.

I want to make sure that the first message would always be created before the second one.